hoot-core/src/main/cpp/hoot/core/elements/Tags.cpp

```diff
@@ -235,6 +236,29 @@ QStringList Tags::getNames() const
   return result;
 }
 
+QStringList Tags::getNamesRecursive() const
+{
```

Comment:
Yeah, I see the source of the confusion. I think just calling getNames() from DuplicateNameRemover will likely do what you want.